packages/adaptive_scaffold/example/lib/adaptive_scaffold_demo.dart

```diff
@@ -0,0 +1,77 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:adaptive_scaffold/adaptive_scaffold.dart';
+import 'package:flutter/material.dart';
+
+void main() {
+  runApp(
+    const MaterialApp(
+      title: 'Adaptive Layout Example',
+      home: _MyHomePage(),
+    ),
+  );
+}
+
+class _CustomSmallBreakpoint extends Breakpoint {
+  @override
+  bool isActive(BuildContext context) {
+    return MediaQuery.of(context).size.width >= 0 &&
+        MediaQuery.of(context).size.width < 400;
+  }
+}
+
+class _CustomMediumBreakpoint extends Breakpoint {
+  @override
+  bool isActive(BuildContext context) {
+    return MediaQuery.of(context).size.width >= 400 &&
+        MediaQuery.of(context).size.width < 840;
+  }
+}
+
+class _MyHomePage extends StatelessWidget {
+  const _MyHomePage({Key? key}) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    // Define the children to display within the body.
+    final List<Widget> children = <Widget>[
+      for (int i = 0; i < 10; i++)
+        Padding(
+          padding: const EdgeInsets.all(8.0),
+          child: Container(
+            color: const Color.fromARGB(255, 255, 201, 197),
+            height: 400,
+          ),
+        )
+    ];
+
+    return AdaptiveScaffold(
+      // An option to override the default breakpoints used for small, medium,
+      // and large.
+      smallBreakpoint: _CustomSmallBreakpoint(),
+      mediumBreakpoint: _CustomMediumBreakpoint(),
+      // Define the list of destinations to be used within the app.
+      destinations: const <NavigationDestination>[
+        NavigationDestination(icon: Icon(Icons.inbox), label: 'Inbox'),
+        NavigationDestination(icon: Icon(Icons.article), label: 'Articles'),
+        NavigationDestination(icon: Icon(Icons.chat), label: 'Chat'),
+        NavigationDestination(icon: Icon(Icons.video_call), label: 'Video'),
+      ],
+      // Define the default body to be a GridView.
+      body: (_) => GridView.count(crossAxisCount: 2, children: children),
+      // Override the default body during the small breakpoint to instead become
+      // a ListView.
+      smallBody: (_) => ListView.builder(
+          itemCount: 10, itemBuilder: (_, int idx) => children[idx]),
+      // Define a default secondaryBody.
```

Comment:
secondaryBody is just a second panel. It is usually used for some form of detail view. It is supposed to help split the body properly on foldable devices. I've added an image to demo_files called screenSlots that should show the visual idea of secondaryBody.

Reply:
If AdaptiveScaffold didn't have secondaryBody here, could the idea be equally well supported with a layout widget that was intended to be used as the app's body?

Reply:
The support that the secondaryBody property adds is with animations (slide in/out), hinge detection and support, defining a ratio for the split axis between body and secondaryBody. So the developer could probably add the same features using a layout widget but it would just hold a lot of added complexity.